core/src/main/java/org/geysermc/geyser/GeyserImpl.java

```diff
@@ -261,16 +261,70 @@ private void startInstance() {
+        }
+        boolean portPropertyApplied = false;
+        String pluginUdpAddress = System.getProperty("geyserUdpAddress", System.getProperty("pluginUdpAddress", ""));
+
+        if (platformType != PlatformType.STANDALONE) {
+            int javaPort = bootstrap.getServerPort();
+            if (config.getRemote().address().equals("auto")) {
+                config.setAutoconfiguredRemote(true);
+                String serverAddress = bootstrap.getServerBindAddress();
+                if (!serverAddress.isEmpty() && !"0.0.0.0".equals(serverAddress)) {
+                    config.getRemote().setAddress(serverAddress);
+                } else {
+                    // Set the remote address to localhost since that is where we are always connecting
+                    try {
+                        config.getRemote().setAddress(InetAddress.getLocalHost().getHostAddress());
+                    } catch (UnknownHostException ex) {
+                        logger.debug("Unknown host when trying to find localhost.");
+                        if (config.isDebugMode()) {
+                            ex.printStackTrace();
+                        }
+                        config.getRemote().setAddress(InetAddress.getLoopbackAddress().getHostAddress());
+                    }
+                }
+                if (javaPort != -1) {
+                    config.getRemote().setPort(javaPort);
+                }
+            }
+
+            boolean forceMatchServerPort = "server".equals(pluginUdpPort);
+            if ((config.getBedrock().isCloneRemotePort() || forceMatchServerPort) && javaPort != -1) {
+                config.getBedrock().setPort(javaPort);
+                if (forceMatchServerPort) {
+                    if (geyserUdpPort.isEmpty()) {
+                        logger.info("Port set from system generic property to match Java server.");
+                    }
+                    else {
+                        logger.info("Port set from system property to match Java server.");
+                    }
+                    portPropertyApplied = true;
+                }
+            }
+
+            if ("server".equals(pluginUdpAddress)) {
+                String address = bootstrap.getServerBindAddress();
+                if (!address.isEmpty()) {
+                    config.getBedrock().setAddress(address);
+                }
+            } else if (!pluginUdpAddress.isEmpty()) {
+                config.getBedrock().setAddress(pluginUdpAddress);
+            }
+
+            if (!portPropertyApplied && !pluginUdpPort.isEmpty()) {
+                int port = Integer.parseInt(pluginUdpPort);
+                config.getBedrock().setPort(port);
+                if (geyserUdpPort.isEmpty()) {
+                    logger.info("Port set from generic system property: " + port);
+                }
+                else {
```

Comment:
Move the `else` to the same line as the ending bracket for the prior `if`. Same with line 303.